collector/meminfo_bsd.go

```diff
@@ -22,30 +22,66 @@ import (
 	"golang.org/x/sys/unix"
 )
 
+type sysctlType uint8
+
+const (
+	t_uint32 sysctlType = iota
+	t_uint64
+)
+
+type meminfoSysctl struct {
+	name       string
+	dataType   sysctlType
+	conversion func(uint64) uint64
+}
+
 func (c *meminfoCollector) getMemInfo() (map[string]float64, error) {
+	var tmp32 uint32
+	var tmp64 uint64
+	var err error
+
 	info := make(map[string]float64)
 
-	size, err := unix.SysctlUint32("vm.stats.vm.v_page_size")
+	tmp32, err = unix.SysctlUint32("vm.stats.vm.v_page_size")
 	if err != nil {
 		return nil, fmt.Errorf("sysctl(vm.stats.vm.v_page_size) failed: %s", err)
 	}
+	size := uint64(tmp32)
+	fromPage := func(v uint64) uint64 {
+		return v * size
+	}
 
-	for key, v := range map[string]string{
-		"active":     "vm.stats.vm.v_active_count",
-		"inactive":   "vm.stats.vm.v_inactive_count",
-		"wire":       "vm.stats.vm.v_wire_count",
-		"cache":      "vm.stats.vm.v_cache_count",
-		"free":       "vm.stats.vm.v_free_count",
-		"swappgsin":  "vm.stats.vm.v_swappgsin",
-		"swappgsout": "vm.stats.vm.v_swappgsout",
-		"total":      "vm.stats.vm.v_page_count",
+	for key, v := range map[string]meminfoSysctl{
+		"active_bytes":         {"vm.stats.vm.v_active_count", t_uint32, fromPage},
+		"inactive_bytes":       {"vm.stats.vm.v_inactive_count", t_uint32, fromPage},
+		"wire_bytes":           {"vm.stats.vm.v_wire_count", t_uint32, fromPage},
+		"cache_bytes":          {"vm.stats.vm.v_cache_count", t_uint32, fromPage},
+		"buffer_bytes":         {"vfs.bufspace", t_uint32, nil},
+		"free_bytes":           {"vm.stats.vm.v_free_count", t_uint32, fromPage},
+		"size_bytes":           {"vm.stats.vm.v_page_count", t_uint32, fromPage},
+		"swap_in_bytes_total":  {"vm.stats.vm.v_swappgsin", t_uint32, fromPage},
+		"swap_out_bytes_total": {"vm.stats.vm.v_swappgsout", t_uint32, fromPage},
+		"swap_size_bytes":      {"vm.swap_total", t_uint64, nil},
```

Comment:
Linux also supports multiple devices, but it's the total swap used that you usually look at in terms of memory. I don't think I've ever seen multiple swap devices in the wild.

This can wait for another PR